libtransact/src/scheduler/multi/shared.rs

```diff
@@ -56,9 +56,15 @@ impl MultiSchedulerShared {
         self.result_callback = Some(callback);
     }
 
+    pub fn batch_already_pending(&self, batch: &BatchPair) -> bool {
+        self.pending_results.contains_key(batch)
+    }
+
     pub fn add_batch(&mut self, batch: BatchPair) {
         self.schedulers.iter_mut().for_each(|scheduler| {
-            scheduler.add_batch(batch.clone());
+            scheduler
+                .add_batch(batch.clone())
+                .expect("Failed to add batch to sub-scheduler");
```

Comment:
In this case, shouldn't the SchedulerError be propagated?

Reply:
I debated whether or not to propagate; I decided not to. The multischeduler already checks for duplicate batches, and its check is going to be more strict than any of the sub-schedulers since it will identify a batch as duplicate until all of the sub-schedulers are done with the batch. Furthermore, because the purpose of the multischeduler is to test the sub-schedulers, if the multischeduler doesn't think a batch is duplicate but a sub-scheduler does, the sub-scheduler is almost definitely broken (so it should fail).

Reply:
We might want to keep the multi-scheduler running in LR networks to observe the number of errors - any expects/panics need to be able to be mitigated to accomplish that later.

Reply:
Fixed to handle errors using the error callback